File: leetcode/solutions/6001.CountingElements/countElements_test.go
package solution

import (
	"testing"
)

func Test_0000_CountingElements(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{nil, 0},
		{[]int{1}, 0},
		{[]int{1, 1, 1}, 0},
		{[]int{1, 2, 3}, 2},
		{[]int{1, 1, 2}, 2},
		{[]int{1, 1, 3, 3, 5, 5, 7, 7}, 0},
		{[]int{1, 3, 2, 3, 5, 0}, 3},
		{[]int{1, 1, 2, 2}, 2},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := countElements(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/1426.CountingElements/countElements_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import countElements from "./countElements.ts";

Deno.test("1426. CountingElements", () => {
  type test = [number[] | null, number];
  const tests: test[] = [
    [null, 0],
    [[1], 0],
    [[1, 1, 1], 0],
    [[1, 2, 3], 2],
    [[1, 1, 2], 2],
    [[1, 1, 3, 3, 5, 5, 7, 7], 0],
    [[1, 3, 2, 3, 5, 0], 3],
    [[1, 1, 2, 2], 2],
  ];

  for (const [arg1, want] of tests) {
    assertEquals(countElements(arg1), want);
  }
});
